bql/semantic/semantic.go

```diff
@@ -75,6 +79,8 @@ type Statement struct {
 	data                      []*triple.Triple
 	pattern                   []*GraphClause
 	workingClause             *GraphClause
+	constructClauses          []*ConstructClause
```

Comment:
Could you elaborate why you need a new working clause? The where clause of a construct is the  same as the one on a regular select. I was wondering if it is just cleaner to just add the actual construct information containing the info to create the new triples. Or am I just mislead by the name?